bluetooth: controller: Mayfly yield after call under Kconfig

mtpr-ot opened the pull request:
Added Kconfig BT_MAYFLY_YIELD_AFTER_CALL to support vendor requirement
of invoking all outstanding mayflies for a given callee in
mayfly_run().

Signed-off-by: Morten Priess <mtpr@oticon.com>

cvinayak commented on subsys/bluetooth/controller/util/mayfly.c:
```diff
@@ -215,6 +215,17 @@ void mayfly_run(u8_t callee_id)
 
+/**
+ * When using cooperative thread implementation, an issue has been seen where
+ * pended mayflies are never exectuted in certain scenarios.
```
executed

cvinayak reviewed (changes requested):
Except for a minor typo otherwise LGTM

mtpr-ot committed:
bluetooth: controller: Mayfly yield after call under Kconfig

Added Kconfig BT_MAYFLY_YIELD_AFTER_CALL to support vendor requirement
of invoking all outstanding mayflies for a given callee in
mayfly_run().

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot commented:
@cvinayak: Could you please approve my change, thanks.

aescolar commented:
@carlescufi Please merge
